File: testdata/jennies/builders/map_index_enum/PHPBuilder/src/Sandbox/SomeStructWithDefaultEnumBuilder.php
<?php

namespace Grafana\Foundation\Sandbox;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\Sandbox\SomeStructWithDefaultEnum>
 */
class SomeStructWithDefaultEnumBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\Sandbox\SomeStructWithDefaultEnum $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\Sandbox\SomeStructWithDefaultEnum();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\Sandbox\SomeStructWithDefaultEnum
     */
    public function build()
    {
        return $this->internal;
    }

    public function data(\Grafana\Foundation\Sandbox\StringEnumWithDefault $key, string $value): static
    {
        $this->internal->data[$key] = $value;
    
        return $this;
    }

}

File: testdata/jennies/builders/map_index_enum/JavaBuilder/sandbox/SomeStructWithDefaultEnumBuilder.java
package sandbox;

import java.util.HashMap;

public class SomeStructWithDefaultEnumBuilder implements cog.Builder<SomeStructWithDefaultEnum> {
    protected final SomeStructWithDefaultEnum internal;
    
    public SomeStructWithDefaultEnumBuilder() {
        this.internal = new SomeStructWithDefaultEnum();
    }
    public SomeStructWithDefaultEnumBuilder data(StringEnumWithDefault key,String value) {
		if (this.internal.data == null) {
			this.internal.data = new HashMap<>();
		}
        this.internal.data.put(key, value);
        return this;
    }
    public SomeStructWithDefaultEnum build() {
        return this.internal;
    }
}
